### PR TITLE
prebid has a whiteboard s2s adapter

### PR DESCRIPTION
**Pull request by jeteve**
This implements a first version of the whiteboard project server side adapter, according to https://trello.com/c/KbeBLyYZ/77-prebid-has-a-whiteboard-s2s-adapter

### CODE REVIEW

File: common/app/conf/switches/CommercialSwitches.scala [hunk 1]
```diff
@@ -387,6 +387,16 @@ trait CommercialSwitches {
     exposeClientSide = true
   )
 
+  val prebidS2SOzoneBidder: Switch = Switch(
```
Comment (kelvin-chappell) [inline common/app/conf/switches/CommercialSwitches.scala]: Is `ozone` the code name?

File: static/src/javascripts/projects/commercial/modules/prebid/bid-config.js [hunk 2]
```diff
@@ -149,6 +151,39 @@ const getImprovePlacementId = (sizes: PrebidSize[]): number => {
...
+                case 'D':
+                    if (containsMpuOrDmpu(sizes)) {
+                        return '13366606';
+                    }
+                    if (containsLeaderboardOrBillboard(sizes)) {
+                        return '13366615';
+                    }
+                    return '';
+                case 'M':
+                    if (contains(sizes, [300, 250])) {
```
Comment (kelvin-chappell) [inline static/src/javascripts/projects/commercial/modules/prebid/bid-config.js]: Perhaps refactor `containsMpuOrDmpu` into `containsMpu or containsDmpu` and use `containsMpu` here?

File: static/src/javascripts/projects/commercial/modules/prebid/prebid.js [hunk 3]
```diff
@@ -28,6 +28,22 @@ const consentManagement = {
...
+const s2sConfig = {
+    accountId: '1',
+    enabled: true,
+    bidders: [
+        'appnexus',
+        'openx', // Defined in the doc.
+    ],
+    timeout: 1000,
+    adapter: 'prebidServer',
+    is_debug: 'true',
```
Comment (kelvin-chappell) [inline static/src/javascripts/projects/commercial/modules/prebid/prebid.js]: Should this be `true` in prod?

File: static/src/javascripts/projects/commercial/modules/prebid/prebid.js [hunk 4]
```diff
@@ -54,19 +70,23 @@ class PrebidAdUnit {
...
-                bidderTimeout,
-                priceGranularity,
-                consentManagement,
-            });
-        } else {
-            window.pbjs.setConfig({
-                bidderTimeout,
-                priceGranularity,
-            });
+            pbjsConfig = Object.assign({}, { consentManagement }, pbjsConfig);
```
Comment (kelvin-chappell) [inline static/src/javascripts/projects/commercial/modules/prebid/prebid.js]: This is better.

File: static/src/javascripts/projects/commercial/modules/prebid/prebid.js [hunk 5]
```diff
@@ -28,6 +28,22 @@ const consentManagement = {
...
 };
 
+const s2sConfig = {
+    accountId: '1',
+    enabled: true,
+    bidders: [
+        'appnexus',
+        'openx', // Defined in the doc.
+    ],
+    timeout: 1000,
```
Comment (kelvin-chappell) [inline static/src/javascripts/projects/commercial/modules/prebid/prebid.js]: Why does the timeout need to be shorter than the auction timeout?

File: static/src/javascripts/projects/commercial/modules/prebid/prebid.js [hunk 6]
```diff
@@ -28,6 +28,22 @@ const consentManagement = {
     allowAuctionWithoutConsent: true,
 };
 
+const s2sConfig = {
+    accountId: '1',
```
Comment (kelvin-chappell) [inline static/src/javascripts/projects/commercial/modules/prebid/prebid.js]: What's this?

File: common/app/conf/switches/CommercialSwitches.scala (hunk 1 above)
Comment (jeteve) [reply common/app/conf/switches/CommercialSwitches.scala]: I'm not sure. The technical project is called the-ozone-project (like in the URL), but the internal codename is 'whitepowder' #suchconfusion

File: static/src/javascripts/projects/commercial/modules/prebid/prebid.js (hunk 5 above)
Comment (jeteve) [reply static/src/javascripts/projects/commercial/modules/prebid/prebid.js]: Don't know. I will change it to 1500 :)

File: static/src/javascripts/projects/commercial/modules/prebid/prebid.js (hunk 6 above)
Comment (jeteve) [reply static/src/javascripts/projects/commercial/modules/prebid/prebid.js]: That's our whitesocks account ID. I couldn't believe myself it was just '1', but the white guys say it is.

File: static/src/javascripts/projects/commercial/modules/prebid/prebid.js (hunk 5 above)
Comment (kelvin-chappell) [reply static/src/javascripts/projects/commercial/modules/prebid/prebid.js]: If it can inherit the same value as the auction that would be great.

**Review by kelvin-chappell — APPROVED** [review]
👍 🏳️

**Pull request merged**